From 5b5a56d154fabe1538ed857dc8ad5147aba94e96 Mon Sep 17 00:00:00 2001
From: Rod Frazer <rfrazer@altera.com>
Date: Mon, 13 Jul 2015 14:01:57 -0400
Subject: [PATCH] altera_gpio_get bit mask error in gpio-altera.c

---
 drivers/gpio/gpio-altera.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-altera.c b/drivers/gpio/gpio-altera.c
index 3aa1176..4a484fc 100644
--- a/drivers/gpio/gpio-altera.c
+++ b/drivers/gpio/gpio-altera.c
@@ -138,7 +138,7 @@ static int altera_gpio_get(struct gpio_chip *gc, unsigned offset)
 {
 	struct of_mm_gpio_chip *mm_gc = to_of_mm_gpio_chip(gc);
 
-	return !!(readl(mm_gc->regs + ALTERA_GPIO_DATA) >> offset);
+	return !!(readl(mm_gc->regs + ALTERA_GPIO_DATA) & BIT(offset));
 }
 
 static void altera_gpio_set(struct gpio_chip *gc, unsigned offset, int value)
-- 
2.1.0

